Enhanced image group by making more flexible for end users.

PascalVorwerk opened the pull request:
# Pull Request

## 📖 Description

Some keys points
- It should be possible to now render a FluentCxImage without a FluentImageGroup. We might want to make this impossible for FluentV5 as the image component might conflict alot with the official library.
- You can apply custom styling to the image, however certain things will not work when inside a group, for example the width and height.. The parent will override the custom width and height to adhere to the groups width and height.
- Border now has to be set by the user, only the border radius is set by the group. (What if the user wants a specific styling of this border?

I intend to now write some unit tests for both the image component and the image group component, but i wanted to check first what you guys think before proceeding with that. I've already added some structure for these tests that will be included in this PR. (Mainly taken from the official library.

If you guys agree with the changes, i can start adding the tests to this same PR soon.

## ✅ Checklist

### General

<!--- Review the list and put an x in the boxes that apply. -->

- [x] I have added tests for my changes. (partially)
- [x] I have tested my changes.

PascalVorwerk commented:
Actually.. it might be annoying for the end user to always having to specify a border for each image.. But i do want it to be customizable.. How should we approach it? @MarvinKlein1508 @JeremyJ78

MarvinKlein1508 commented:
Personally what I would do is renaming the component to `FluentCxImageGroupImage` or `FluentCxImageGroupItem` to make it only usable in this component. As you said in V5 this component will be there out of the box and creating a custom component for images in v4 can be done pretty easily if you need really need it in your project.

PascalVorwerk commented:
> Personally what I would do is renaming the component to FluentCxImageGroupImage or FluentCxImageGroupItem to make it only usable in this component. As you said in V5 this component will be there out of the box and creating a custom component for images in v4 can be done pretty easily if you really need it in your project.

okay, i do agree, i have renamed the class and added a exception to the onitialized whenever a user tries to use the component outside of the group component.

PascalVorwerk commented:
I have added 2 more properties:

- BackgroundStyle: Allow to set 1 styling for the entire group, but it will be overriden on individual items if they have their own styling..
- BorderStyle: The same as above, but then for border.. 

Also i renamed the component to FluentImageGroupItem, let me know what you guys think. If this is a approach we can agree on, i will start creating some tests for this component in this PR

PascalVorwerk commented:
I upped the coverage from 0% to 2% 💀
